doc/developers/advanced_installation.rst

```diff
@@ -75,7 +83,7 @@ feature, code or documentation improvement).
 
      python3 -m venv sklearn-env
      source sklearn-env/bin/activate
-     pip install wheel numpy scipy cython
+     pip install -r requirements/common.txt
```

Comment:
I remember that scikit-image uses a `requirements.txt` on the top level that calls the sub-requirements: https://github.com/scikit-image/scikit-image/blob/main/requirements.txt

I would prefer something like that indeed. It would be easier to keep the package up-to-date with minimal editing.

Reply:
To that end an important question is do we want a monothilic installation -- we recommend every contributor install every potentially necessary package (union of `common, dev, doc`) or do we allow users the flexibility of choosing their own as needed?

I prefer the flexible approach becasue, for example, the `doc` packages are large (`>100Mb` I believe) and some users on constrained systems may appreciate the option to do without them.

If we choose monolithic, it certainly is easy enough to have a file that refers to these specific sub-groups as you mentioned. We could even have a monolithic `.yml` environment file (although I think we would just need to manually specify packages, not other files).

We could also choose to maintain both? Though this seems a bit complex for not much gain. I think we should instead decide one way or another.

Reply:
We can still have the `requirements.txt` that installs all of the sub-packages but still request to install the `requirements/build.txt` for installing from the source.

The top-level `requirements.txt` would be useful in the "Contributing Guideline" where you want everyone to install `build`, `dev`, and `doc`.